thoth/license_solver/comparator.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# license-solver
+# Copyright(C) 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A Class compare classifier and license."""
+
+import os
+import re
+import yaml
+import attr
+import logging
+from typing import List, Any, Dict
+from .package import Package
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _delete_brackets(license_list: str) -> str:
+    return re.sub(r"(\(?)(\)?)", "", license_list).strip()
+
+
+def _delete_brackets_and_content(license_list: str) -> str:
+    return re.sub(r"\(.*?\)", "", license_list).strip()
+
+
+@attr.s(slots=True)
+class Comparator:
+    """Class Comparator compare classifiers and licenses."""
+
+    _comparator_dictionary: Dict[str, Any] = attr.ib(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        """Open dictionary for comparing license and classifier."""
+        file_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data/comparator_dictionary.yaml")
+        with open(file_path) as f:
+            try:
+                self._comparator_dictionary = yaml.safe_load(f)
+            except yaml.YAMLError:
+                _LOGGER.warning("Can't open data/comparator_dictionary.yaml or broken file")
+                exit(1)
```

Comment:
If this tool will be used as a library, it might be better to raise an exception here not to halt the whole program that uses this functionality.